### PR TITLE
Added tasks 566, 567, 572

### PR DESCRIPTION
**Pull request by ThanhNIT**

### CODE REVIEW

**Comment by javadev** [comment]
We need more tests for task 572

image

**Comment by ThanhNIT** [comment]
> We need more tests for task 572
> 
> image

File: src/test/java/g0501_0600/s0572_subtree_of_another_tree/SolutionTest.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.Arrays;
+import org.junit.jupiter.api.Test;
+
+class SolutionTest {
+    @Test
+    void isSubtreeFound() {
+        TreeNode treeNode =
+                TreeUtils.constructBinaryTree(new ArrayList<>(Arrays.asList(3, 4, 5, 1, 2)));
+        TreeNode subTree = TreeUtils.constructBinaryTree(new ArrayList<>(Arrays.asList(4, 1, 2)));
+        assertThat(new Solution().isSubtreeFound(treeNode, subTree), equalTo(false));
```
Comment (javadev) [inline src/test/java/g0501_0600/s0572_subtree_of_another_tree/SolutionTest.java:18]: It should be true

image

File: src/test/java/g0501_0600/s0572_subtree_of_another_tree/SolutionTest.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import com_github_leetcode.TreeUtils;
+import java.util.ArrayList;
+import java.util.Arrays;
+import org.junit.jupiter.api.Test;
+
+class SolutionTest {
+    @Test
+    void isSubtreeFound() {
+        TreeNode treeNode =
+                TreeUtils.constructBinaryTree(new ArrayList<>(Arrays.asList(3, 4, 5, 1, 2)));
```
Comment (javadev) [inline src/test/java/g0501_0600/s0572_subtree_of_another_tree/SolutionTest.java:16]: We don't need `new ArrayList<>(` wrapper.

Comment (ThanhNIT) [reply src/test/java/g0501_0600/s0572_subtree_of_another_tree/SolutionTest.java:16]: I will update

**Comment by javadev** [comment]
✔

File: src/test/java/g0501_0600/s0572_subtree_of_another_tree/SolutionTest.java (hunk 2 above)
Comment (javadev) [reply src/test/java/g0501_0600/s0572_subtree_of_another_tree/SolutionTest.java:16]: I prepared a fix.

Comment (ThanhNIT) [reply src/test/java/g0501_0600/s0572_subtree_of_another_tree/SolutionTest.java:16]: thanks

**Pull request merged**